### PR TITLE
Don't error on missing `onTick` method in Lua

### PR DESCRIPTION
**Pull request by stieg**
This change alters our Lua subsystem to no longer error on a
missing `onTick` method.  This will allow users to create scripts
that run one time during first load of the script and will also
set us up to be able to fire callbacks in future versions should
we choose to do so.

Issue #707

### CODE REVIEW

File: src/lua/luaTask.c [hunk 1]
```diff
@@ -166,9 +166,10 @@ static int lua_invocation(struct lua_run_state *rs)
         /* Now run the callback */
         lua_getglobal(rs->lua_state, function);
         if (lua_isnil(rs->lua_state, -1)) {
-                pr_error_str_msg(_LOG_PFX "Function not found: ", function);
+                /* No longer a failure per Issue #707 */
+                pr_debug_str_msg(_LOG_PFX "Function not found: ", function);
                 lua_pop(rs->lua_state, 1);
-                status = LUA_ERR_CALLBACK_NOT_FOUND;
```
Comment (brentpicasso) [inline src/lua/luaTask.c]: should we remove this #define since if we're not using it elsewhere?

File: src/lua/luaTask.c [hunk 2]
```diff
@@ -166,9 +166,10 @@ static int lua_invocation(struct lua_run_state *rs)
...
+                status = 0;
```
Comment (brentpicasso) [inline src/lua/luaTask.c]: should we add a #define for indicating "no lua error"?

File: src/lua/luaTask.c (hunk 1 above)
Comment (stieg) [inline src/lua/luaTask.c]: Yeah was debating that myself.  Can't hurt.

File: src/lua/luaTask.c (hunk 2 above)
Comment (stieg) [inline src/lua/luaTask.c]: Ok

**Comment by brentpicasso** [comment]
:shipit:

**Pull request merged**